modules/core/src/main/scala/Trace.scala

```diff
@@ -86,4 +86,18 @@ object Trace {
 
   }
 
+  implicit def resourceInstance[F[_]: Bracket[?[_], Throwable]](implicit T: Trace[F]): Trace[Resource[F, ?]] =
+    new Trace[Resource[F, ?]] {
+
+      override def put(fields: (String, TraceValue)*): Resource[F, Unit] =
+        Resource.liftF(T.put(fields: _*))
+
+      override def kernel: Resource[F, Kernel] =
+        Resource.liftF(T.kernel)
+
+      override def span[A](name: String)(fa: Resource[F, A]): Resource[F, A] =
+        Resource(T.span(name)(fa.allocated))
```

Comment:
Hm, it seems like we want the extent of the span to be `use` rather than `allocated`.

Reply:
I'm not sure that `fa.use(...)` can be used because there is no available `A => F[B]`  function for `use` method. 

I've added some tests. Looks like this implementation works correctly.